### DISCUSSION TITLE
Mac ARM64/Apple Silicon Binary Contribution?

### DISCUSSION THREAD

**Discussion by icase81** [supermerill/SuperSlicer#3692] [Community contribution]
I compiled the latest release last night on my Mac M1. Everything went fine and works just fine, but I figured if I could contribute the binary to add it might help some other people out there. Is there any way to do this?

**Comment by quattroerik** [comment]
Would love to see that :-) I don't have time to dig into this topic.
What is strange for me, i have just downloaded the latest MAC (not ARM) version and i can start it on my M1 Mac Mini. But im not sure if that is right?

**Reply by icase81** [reply]
Apple Silicon Macs can run Intel versions or programs, there's just a performance hit because it has to run through the Rosetta emulation layer.

**Reply by quattroerik** [reply]
Thanks, happy to use the new version. And to be frank i do not feel its slow on my system :-)

**Reply by stanpete78** [reply]
I would also love to see that. Did you find out how to upload it yet?

**Reply by sammcj** [reply]
Did you ever find a way to contribute your 2.5 builds @icase81? I'm assuming @supermerill needs to approve and pull it into the repo / release.

**Reply by salbang** [reply]
> Did you ever find a way to contribute your 2.5 builds @icase81? I'm assuming @supermerill needs to approve and pull it into the repo / release.

I have raised an issue here: https://github.com/supermerill/SuperSlicer/issues/3762
I may request pull, just don't have enough time because of my work for living :-(

**Comment by HelloThisIsFlo** [comment]
Side question @icase81 , how did you manage to do that? 

I fixed every dependency problem I came across but then I got this error and I didn't know what to do next
```
CMake Error at src/libslic3r/CMakeLists.txt:324 (find_package):
  By not providing "FindCGAL.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "CGAL", but
  CMake did not find one.

  Could not find a package configuration file provided by "CGAL" with any of
  the following names:

    CGALConfig.cmake
    cgal-config.cmake

  Add the installation prefix of "CGAL" to CMAKE_PREFIX_PATH or set
  "CGAL_DIR" to a directory containing one of the above files.  If "CGAL"
  provides a separate development package or SDK, be sure it has been
  installed.
  ```
  
  Did you have the same, or it just worked? 🤔

**Reply by ZyberSE** [reply]
If someone could share a compiled binary it would be great.

**Reply by salbang** [reply]
> Side question @icase81 , how did you manage to do that?
> 
> I fixed every dependency problem I came across but then I got this error and I didn't know what to do next
> 
> ```
> CMake Error at src/libslic3r/CMakeLists.txt:324 (find_package):
>   By not providing "FindCGAL.cmake" in CMAKE_MODULE_PATH this project has
>   asked CMake to find a package configuration file provided by "CGAL", but
>   CMake did not find one.
> 
>   Could not find a package configuration file provided by "CGAL" with any of
>   the following names:
> 
>     CGALConfig.cmake
>     cgal-config.cmake
> 
>   Add the installation prefix of "CGAL" to CMAKE_PREFIX_PATH or set
>   "CGAL_DIR" to a directory containing one of the above files.  If "CGAL"
>   provides a separate development package or SDK, be sure it has been
>   installed.
> ```
> 
> Did you have the same, or it just worked? 🤔

I got the same error, but after resolving that by brew install cgal, i get...
CMake Error at /opt/homebrew/Cellar/cmake/3.26.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find wxWidgets (missing: wxWidgets_LIBRARIES) (Required is at
  least version "3.1")
Call Stack (most recent call first):
  /opt/homebrew/Cellar/cmake/3.26.1/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /opt/homebrew/Cellar/cmake/3.26.1/share/cmake/Modules/FindwxWidgets.cmake:977 (find_package_handle_standard_args)
  src/CMakeLists.txt:58 (find_package)

I don't know how to resolve this...

**Reply by icase81** [reply]
Sorry, it just worked for me. Give me a few minutes and I'll fork the main repo and upload a compiled binary. I'll update here once complete.

**Reply by salbang** [reply]
I could build the master branch, but I had to change the hash value for TIFF dependency.
In .../SuperSlicer/deps/TIFF$ ll
total 8
-rw-r--r--  1 jjung  staff  950 Mar 27 23:09 TIFF.cmake
Edited the file so that the hash value becones
        URL https://gitlab.com/libtiff/libtiff/-/archive/v4.3.0/libtiff-v4.3.0.zip
        URL_HASH SHA256=a1db6826ea1b8b08eaf168973abb29b8a143fb744c1b24cc6967cb12f5e6e81f

**Reply by salbang** [reply]
I could build the master_dev branch with the same modification.
https://1drv.ms/u/s!As148oJiKUy1hJYdIa_mgkInEu7jlA?e=OrsGEZ
This is the 2.5.59 version for Apple Silicon. Compiled in MacMini M2 Pro

**Reply by sammcj** [reply]
@salbang your 2.5 build link seems to be broken - any chance of an updated one? (or @icase81)

**Reply by salbang** [reply]
Check with this link.
https://1drv.ms/u/s!As148oJiKUy1hJYdIa_mgkInEu7jlA?e=PG6FcK

**Reply by sammcj** [reply]
Legend! That worked, thank you!

**Reply by salbang** [reply]
Just made the pull request for the hash value fix for TIFF.cmake.

**Comment by icase81** [comment]
Recompiled just now. Located at:

https://github.com/icase81/SuperSlicerAppleSilicon/releases/tag/MacM1

**Reply by stanpete78** [reply]
You, sir, are awesome. Thanks.

**Reply by ZyberSE** [reply]
Thank you, I've downloaded it twice but it says the file is damaged and can't be opened?
Extracted size is 107mb (if I remember correct)

**Reply by stanpete78** [reply]
> Thank you, I've downloaded it twice but it says the file is damaged and can't be opened?
> Extracted size is 107mb (if I remember correct)

It works for me. Open Terminal.app and do a "xattr -d com.apple.quarantine /Applications/Superslicer.app"

**Reply by ZyberSE** [reply]
That resolved it, thanks! Awesome to have the 3d space mouse working again :)

**Reply by salbang** [reply]
I found that the compiled version is from master branch which is 2.4 version.

**Reply by ZyberSE** [reply]
Yeah I just redownloaded to confirm and as you say, the version number matches the master branch exactly, so this does not seam to be v2.5?

**Reply by salbang** [reply]
> Yeah I just redownloaded to confirm and as you say, the version number matches the master branch exactly, so this does not seam to be v2.5?

No, it is not 2.5. 2.4, indeed. I've shared 2.5 from master_dev in my one drive. Link is in my previous comment.
